interface/Yami.h

```diff
@@ -18,11 +18,18 @@
 #define YAMI_H_
 
 #include <YamiVersion.h>
+#include <VideoCommonDefs.h>
 
+#if (__BUILD_DECODER__)
```

Comment:
It is not a good idea to do this in a public header.  It forces clients to define these also.  Also, if client project defines `__BUILD_DECODER__` (etc...) and libyami was compiled without `__BUILD_DECODER__` (etc...), then problems occur.  Thus, it's better not to impose these complexities onto clients.

Perhaps you could generate Yami.h at libyami compile time to insert appropriate includes instead.

Reply:
Thanks for your review and helpful suggestion @uartie ;
But here one more explaination about this piece of code, the macros like __BUILD_DECODER__ are defined in the file YamiVersion.h, clients not have to define them, it can be opened or closed according to the compilation options.
Yes, I agree with you about your suggestion, generation Yami.h is better.

Reply:
Ah yes, I missed those defines in YamiVersion.h.  Thanks for pointing them out.  In that case, the solution is okay for me.  The only thing I would change are the names of the defines; `__BUILD_DECODER__` (etc...) are too generic.  See my suggestion in YamiVersion.h comment.

Reply:
Of course, if generating Yami.h... then you could:
```
#if @__BUILD_DECODER__@
#include <VideoDecoderHost.h>
#endif
```
...to eliminate the `#defines` altogether.